src/core/ConfigManager.cpp

```diff
@@ -400,11 +403,21 @@ void ConfigManager::loadConfigFile(const QString & configFile)
 
 			QDomNode node = root.firstChild();
 
-			// Cache the config version for upgrade()
+			// Cache LMMS version
 			if (!root.attribute("version").isNull()) {
 				m_version = root.attribute("version");
 			}
 
+			// Get the version of the configuration file (for upgrade purposes)
+			if( root.attribute("configversion").isNull() )
+			{
+				m_configVersion = -1; // No configversion attribute found
```

Comment:
I think you should set `m_configVersion = configVersion();` here instead, and rework `configVersion()` as described in my comment there.